internal/confmapprovider/discovery/discoverer.go

```diff
@@ -246,11 +250,7 @@ func (d *discoverer) createObserver(observerID component.ID, cfg *Config) (otelc
 	}
 
 	if ce := d.logger.Check(zap.DebugLevel, "unmarshalled observer config"); ce != nil {
-		if c, e := yaml.Marshal(observerConfig); e != nil {
-			d.logger.Debug("failed marshaling observer config for logging", zap.Error(e))
-		} else {
-			ce.Write(zap.String("config", string(c)))
-		}
+		ce.Write(zap.String("config", fmt.Sprintf("%#v\n", observerConfig)))
```

Comment:
This fixes a bug where I was trying to marshal unexported structs so we're using their value representation instead.